### PR TITLE
Fix iproto index name to ID mapping

### LINKED ISSUE #9923: Invalid IPROTO_INDEX_NAME mapping on the server side.
**Bug description**

The server could not correctly map an index from a string passed with IPROTO_INDEX_NAME to a number.
I found the problem with the `tarantool/go-tarantool` connector, but I'm able to reproduce it with `net.box`.

* OS: Arhclinux
* Architecture: amd64

Place `tarantool --version` output here.

```
$ tarantool --version
Tarantool 3.1.0-entrypoint-214-ge048a5021
Target: Linux-x86_64-Debug
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=TRUE
Compiler: GNU-13.2.1
C_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/home/ozi/src/tarantool=. -std=c11 -Wall -Wextra -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type -Werror -g -ggdb -O0 
CXX_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -msse2 -Wformat -Wformat-security -Werror=format-security -fstack-protector-strong -fPIC -fmacro-prefix-map=/home/ozi/src/tarantool=. -std=c++11 -Wall -Wextra -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type -Werror -g -ggdb -O0
```

**Steps to reproduce**

1. Run the script with `tarantool -i`:
```Lua
box.cfg({
    listen = "localhost:3013",
})
box.once("init", function()
    local s = box.schema.space.create('test', {
        id = 621,
        temporary = true,
        if_not_exists = true,
        field_count = 3,
        format = {
            {name = "id", type = "unsigned"},
            {name = "name", type = "string"},
            {name = "arr1", type = "array"},
        },
    })
    s:create_index('primary', {
        type = 'tree',
        unique = true,
        parts = {1, 'unsigned'},
        if_not_exists = true
    })
    s:create_index('secondary', {
        id = 5, type = 'tree',
        unique = false,
        parts = {2, 'string'},
        if_not_exists = true
    })

    box.schema.user.create('test', {password = 'test'})
    box.schema.user.grant('test', 'execute', 'universe')
    box.schema.user.grant('test', 'read,write', 'space', 'test')
end)
```
2. Run a Tarantool in the second console and execute:
```
tarantool> conn = require('net.box').connect("localhost:3013", {user = 'test', password = 'test', fetch_schema = false})
---
...

tarantool> conn.space.test.index.secondary:select()
---
- error: 'No index #1 is defined in space ''test'''
...

tarantool> conn.space.test.index[5]:select()
---
- []
...
```

**Actual behavior**

The server does not correctly map an index id by an index name passed with `IPROTO_INDEX_NAME`.

**Expected behavior**

The server selects data from an index with a name passed with `IPROTO_INDEX_NAME`.

### PR DESCRIPTION
**Pull request by rorororom**
iproto: fix IPROTO index name to ID mapping

Previously, index names were resolved to dense IDs, which caused
failures when accessing indexes by name. This happened because the
index_map lookup expected user-visible IDs (index->def->iid), not dense IDs.

The fix replaces dense_id with index->def->iid to ensure correct mapping
of index names to their numeric IDs, aligning with the expected behavior
when accessing indexes by name.

Closes #9923

### CODE REVIEW

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        })
+
+        local idx_name_result = s.index.secondary:select()
+        t.assert_equals(#idx_name_result, 0)
+
+        local idx_id_result = s.index.primary:select()
+        t.assert_equals(#idx_id_result, 0)
+
+    end)
+end
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:45]: The test doesn't cover anything - it works fine on master. Please, check that the new tests actually fail before the fix is applied.

Also:
1. The commit fixes IPROTO resolver - here you use indexes locally.
2. The problem was caused by sparse index id. Here are no spare index ids.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+g.before_all(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+end)
+
+g.after_all(function()
+    g.server:stop()
+end)
+
+g.test_index_by_name = function()
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:15]: The helper that you fixed is used in `tx_process1` - let's cover it with a case as well. Using `space.index.secondary:delete` would be sufficient.

**Review by drewdzzz — CHANGES_REQUESTED** [review]
Thanks for the patch! See my comments below.

Regarding commit message:

Header:
1. Do not use capital letter after the category name (`Correct` -> `correct`)
2. Let's replace word "correct" with "fix" - "correct" is often used as an adjective and "fix" is always a verb.

Body:
1. Please, do not use multiple spaces between sentences.
2. I would rephrase it in such way: explain the problem (that Tarantool resolved name to dense id but Tarantool requests actually expect id observed by user - `index->def->iid`) and after that something like "The commit fixed this problem". What's about the current message - it's still not clear why you replaced `dense_id` with `index->def->iid`.
3. "fail due to incorrect index ID resolution when fetch_schema was false" - did you check it? Does it really works with `fetch_schema = true`?

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.test_index_access_errors = function()
+    -- fetch schema = false to resolve index on server instead of client
+    local conn = net.connect(g.server.net_box_uri, {fetch_schema = false})
+    t.assert_is_not(conn, nil)
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:52]: Let's move connection creation to `before_each` helper. You can do it in this way:
```lua
g.conn = net.connect(...)
```

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    local space = conn.space.test
+    t.assert_is_not(space, nil)
+
+    local res1 = space.index.secondary:select()
+    local res2 = space.index[5]:select()
+    t.assert_equals(res1, res2)
+
+    conn:close()
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:61]: Ditto - let's move it to `after_each`.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+g.test_index_delete = function()
+    local conn = net.connect(g.server.net_box_uri, {fetch_schema = false})
+    t.assert_is_not(conn, nil)
+
+    local space = conn.space.test
+    t.assert_is_not(space, nil)
+
+    space:insert({2, 'A', {1, 2, 3}})
+    space:insert({4, 'B', {4, 5, 6}})
+    space:insert({8, 'C', {7, 8, 9}})
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:73]: Since you insert some tuples for test, let's do `space:truncate` in `after_each` hook so that these insertions won't affect other cases.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            if_not_exists = true,
+        })
+    end)
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.test_index_access_errors = function()
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:49]: The second case checks access errors too. Let's rename this case to reflect that we check `select` here.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    space:insert({4, 'B', {4, 5, 6}})
+    space:insert({8, 'C', {7, 8, 9}})
+
+    local deleted_by_name = space.index.unique_secondary:delete('A')
+    t.assert_equals(deleted_by_name, {2, 'A', {1, 2, 3}})
+
+    local check_deleted_by_name = space:get(2)
+    t.assert_is(check_deleted_by_name, nil)
+
+    local deleted_by_id = space.index[6]:delete('C')
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:81]: Why do we need deletion by id here?

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+            if_not_exists = true,
+        })
+        box.space.test:create_index('secondary', {
+            id = 5,
+            type = 'tree',
+            unique = false,
+            parts = {2, 'string'},
+            if_not_exists = true,
+        })
+        box.space.test:create_index('unique_secondary', {
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:35]: Why do you need both unique and non-unique indexes here? Let's just add `unique = true` to `secondary`.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    cg.server:exec(function()
+        box.schema.create_space('test', {
+            id = 621,
+            temporary = true,
+            if_not_exists = true,
+            field_count = 3,
+            format = {
+                {name = "id", type = "unsigned"},
+                {name = "name", type = "string"},
+                {name = "arr1", type = "array"},
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:19]: Do you actually need this `arr1` field? Let's drop it to simplify the test.
Also, let's drop `field_count` and `temporary` attributes and explicit id of the space - they have nothing in common with bug we are fixing.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+    cg.server:exec(function()
+        box.schema.create_space('test', {
+            id = 621,
+            temporary = true,
+            if_not_exists = true,
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:14]: Since you create the space and indexes in `before_all`, you don't need the `if_not_exists` attribute. Let's drop all `if_not_exists` attributes.

**Review by drewdzzz — CHANGES_REQUESTED** [review]
Thanks for the fixes!
I have no objections regarding the fix, commit message and changelog. Let's just slightly improve the test.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 3 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 4 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 5 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 6 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 7 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 9 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 10 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.before_each(function(cg)
+    cg.server:exec(function()
+        box.schema.create_space('test', {
+            id = 621,
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:19]: Let's drop this id - it's not actually needed.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+            parts = {1, 'unsigned'},
+        })
+        box.space.test:create_index('secondary', {
+            id = 5,
+            type = 'tree',
+            unique = true,
+            parts = {2, 'string'},
+        })
+    end)
+    cg.conn = net.connect(cg.server.net_box_uri, {fetch_schema = false})
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:37]: Why did you drop the comment why we need `fetch_schema = false`?

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+    end)
+end)
+
+g.test_index_select = function(cg)
+    local space = cg.conn.space.test
+    t.assert_is_not(space, nil)
+
+    local res1 = space.index.secondary:select()
+    local res2 = space.index[5]:select()
+    t.assert_equals(res1, res2)
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:56]: Let's insert some tuples to the space along the way - there is no point to compare results since they are empty.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.before_each(function(cg)
+    cg.server:exec(function()
+        box.schema.create_space('test', {
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:18]: Let's create the space and indexes in `before_all` - in `after_each` hook you can call `space:truncate()` to drop all tuples.

**Review by drewdzzz — CHANGES_REQUESTED** [review]
Thanks for the fixes! Looks good, just a few nitpicks.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    cg.server:drop()
+end
+
+g.before_each = function(cg)
+    cg.server:exec(function()
+        box.space.test:truncate()
+    end)
+end
+
+g.after_each = function(cg)
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:47]: These hooks are not actually executed because they are defined in this way:
```lua
g.after_each = function() ... end
```

Please, define them in this way:
```lua
g.after_each(function() ... end)
```

Applicable to all hooks: `before_all`, `after_all`, `before_each`, `after_each`.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+        box.space.test:create_index('secondary', {
+            id = 5,
+            type = 'tree',
+            unique = true,
+            parts = {2, 'string'},
+        })
+    end)
+
+    -- fetch schema = false to resolve index on server instead of client
+    cg.conn = net.connect(cg.server.net_box_uri, {fetch_schema = false})
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:33]: You should create the connection in `before_each` hook instead of `before_all` since you drop it in `after_each`. Since `after_each` is not actually executed (see another comment), it doesn't break anything, but it will after the fix.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    cg.conn:close()
+end
+
+g.test_index_select = function(cg)
+    local space = cg.conn.space.test
+    t.assert_is_not(space, nil)
+
+    space:insert({1, 'F'})
+
+    local res1 = space.index.secondary:select()
```
Comment (drewdzzz) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:57]: Let's do `index:select('F')` to check if select was actually done by a string field.

**Review by drewdzzz — CHANGES_REQUESTED** [review]
Thanks for the fixes! Almost perfect, only a few remarks.

**Review by drewdzzz — APPROVED** [review]
Thanks for the patch, LGTM!

File: changelogs/unreleased/gh-9923-fix-invalid-iproto-name-mapping-on-the-server-side.md [hunk 18]
```diff
@@ -0,0 +1,3 @@
+## bugfix/core
+
+ * Fixed a bug when IPROTO_INDEX_NAME was mapped incorrectly into index id (gh-9923).
```
Comment (CuriousGeorgiy) [inline changelogs/unreleased/gh-9923-fix-invalid-iproto-name-mapping-on-the-server-side.md:3]: Nit:

```suggestion
 * Fixed a bug when IPROTO_INDEX_NAME was mapped into a wrong index identifier (gh-9923).
```

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+
+g.test_index_select = function()
+    local space = g.conn.space.test
+    t.assert_is_not(space, nil)
+
+    space:insert({1, 'F'})
+
+    local res1 = space.index.secondary:select('F')
+    local res2 = space.index[5]:select('F')
+    t.assert_equals(res1, res2)
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:59]: Please add another assert that one of the results is equal to `{{1, f}}`. Otherwise, `res1` and `res2` could both return wrong results and you would not catch it.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+end
+
+g.test_index_delete = function()
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:62]: And where is `test_index_get`? Actually, I believe it is enough to test only one method, since they all go through the same code path, but if you want to cover everything, then please cover everything.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+    local space = g.conn.space.test
+    t.assert_is_not(space, nil)
+
+    space:insert({2, 'A'})
+    space:insert({4, 'B'})
+    space:insert({8, 'C'})
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:68]: What's the point of inserting other values here? Please keep your tests concise and straight to the point, so that a reviewer and future reader of your test does not spend time guessing why this is needed.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+
+    local deleted_by_name = space.index.secondary:delete('A')
+    t.assert_equals(deleted_by_name, {2, 'A'})
+
+    local check_deleted_by_name = space:get(2)
+    t.assert_is(check_deleted_by_name, nil)
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:74]: Please also add a check for the `get` from the secondary index by name.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+end)
+
+g.after_all(function()
+    g.server:drop()
+end)
+
+g.before_each(function()
+    -- fetch schema = false to resolve index on server instead of client
+    g.conn = net.connect(g.server.net_box_uri, {fetch_schema = false})
+    t.assert_is_not(g.conn, nil)
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:40]: This check is redundant. Your connection can never be `nil`. If an error occurs during the connection construction, an exception is thrown. Since you do not specify the `wait_connected` option (which is totally fine for a test), you are by default blocked until the connection is established [^1], so you do not have to care about this:
> wait_connected: a connection timeout. By default, the connection is blocked until the connection is established, but if you specify wait_connected=false, the connection returns immediately. If you specify this timeout, it will wait before returning (wait_connected=1.5 makes it wait at most 1.5 seconds).

[^1]: https://www.tarantool.io/en/doc/latest/reference/reference_lua/net_box/#lua-function.net_box.connect

**Review by CuriousGeorgiy — APPROVED** [review]
Hey Maria, nice to meet you! Great first patch to start your Tarantool journey!

I have left some comments below regarding polishing of the test. These comments are mostly nitpicks, lifehacks, tricks and conventional Tarantool wisdom.

Please expect a lot of them in the beginning of your journey — they do not in any way diminish your hard skills, their goal is rather to calibrate you to our development traditions and expectations.

**Review by xuniq — APPROVED** [review]

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 22 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:73]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 20 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:63]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 22 above)
Comment (CuriousGeorgiy) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:73]: Look's like you forgot to push the change.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+    end)
+end)
+
+g.after_all(function()
+    g.server:drop()
+end)
+
+g.before_each(function()
+    -- fetch schema = false to resolve index on server instead of client
+    g.conn = net.connect(g.server.net_box_uri, {fetch_schema = false})
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:39]: Nit: do you really need to recreate the connection after each test? Maybe do it once in `before_all`/`after_all`?

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+
+g.before_all(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+    g.server:exec(function()
+        box.schema.create_space('test', {
+            format = {
+                {name = "id", type = "unsigned"},
+                {name = "name", type = "string"},
+            },
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:15]: Nit: do you actually rely on the space format in the test? I think you can safely drop this as redundant.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+        })
+
+        box.space.test:create_index('primary', {
+            type = 'tree',
+            unique = true,
```
Comment (CuriousGeorgiy) [inline test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:20]: Nit: `type = 'tree'` and `unique = true` are defaults that everyone recognizes, so you can safely drop them as redundant.

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 22 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua:58]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 24 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+    local deleted_by_name = space.index.secondary:delete('A')
+    t.assert_equals(deleted_by_name, {2, 'A'})
+
+    local check_deleted_by_name = space:get(2)
+    t.assert_is(check_deleted_by_name, nil)
+end
+
+g.test_index_get = function()
+    local space = g.conn.space.test
+    t.assert_is_not(space, nil)
```
[the remark this answers is not in the document]
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

File: test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua (hunk 26 above)
Comment (rorororom) [reply test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua]: fix

**Comment by sergepetrenko** [comment]
Backports:
3.3 https://github.com/tarantool/tarantool/pull/10982
3.2 https://github.com/tarantool/tarantool/pull/10983
